library/network/src/modules/SuSEFirewall.rb

```diff
@@ -83,8 +100,54 @@ def self.get_running_backends
       running_backends
     end
 
+    # Create appropriate firewall instance based on factors such as which backends
+    # are available and/or running/selected.
+    # @return SuSEFirewall2 or SuSEFirewalld instance.
     def self.create
-      SuSEFirewall2.new
+      Yast.import "Mode"
+
+      # Old testsuite
+      if Mode.testsuite
```

Comment:
Checking `Mode.testsuite` should be avoided.
OTOH I see that existing code in this file does it too so it may be hard to get rid of :-/